1-js/04-object-basics/07-optional-chaining/article.md

```diff
@@ -1,176 +1,175 @@
 
-## Optional chaining
+## Encadeamento opcional
 
-The optional chaining `?.` stops the evaluation and returns `undefined` if the part before `?.` is `undefined` or `null`.
+O encadeamento opcional `?.` acaba com a avaliação e retorna `undefined` se a parte anterior de `?.` é `undefined` ou `null`.
 
-Further in this article, for brevity, we'll be saying that something "exists" if it's not `null` and not `undefined`.
+Ao longo deste artigo, por uma questão de tempo, diremos que algo "existe" se não é `null` e não é `undefined`.
 
 
-Here's the safe way to access `user.address.street`:
+Aqui está a forma segura de acessar `user.address.street`:
 
 ```js run
-let user = {}; // user has no address
+let user = {}; // o usuário não possui um endereço
 
-alert( user?.address?.street ); // undefined (no error)
+alert( user?.address?.street ); // indefinido (sem erro)
 ```
 
 Reading the address with `user?.address` works even if `user` object doesn't exist:
+Ler o endereço com `user?.address` funciona mesmo se o objeto `user` não existir:
 
 ```js run
 let user = null;
 
-alert( user?.address ); // undefined
+alert( user?.address ); // indefinido
 
-alert( user?.address.street ); // undefined
-alert( user?.address.street.anything ); // undefined
+alert( user?.address.street ); // indefinido
+alert( user?.address.street.anything ); // indefinido
 ```
 
-Please note: the `?.` syntax works exactly where it's placed, not any further.
+Observação: A sintaxe `?.` funciona exatamente aonde ela é colocada, não adiante.
 
-In the last two lines the evaluation stops immediately after `user?.`, never accessing further properties. But if the `user` actually exists, then the further intermediate properties, such as `user.address` must exist.
+Nas últimas duas linhas a avaliação para imediatamente após `user?.`, não chegando a acessar as seguintes propriedades. Mas se `user` de fato existir, então as propriedades seguintes, como `user.address` devem existir.
 
-```warn header="Don't overuse the optional chaining"
-We should use `?.` only where it's ok that something doesn't exist.
+```warn header="Não abuse do encadeamento opcional"
+Nós devemos usar `?.` apenas aonde é possível ter algo que não existe.
 
-For example, if according to our coding logic `user` object must be there, but `address` is optional, then `user.address?.street` would be better.
+Por exemplo, se de acordo com a nossa lógica de código o objeto `user` existe, porém `address` é opcional, então seria melhor utilizar `user.address?.street`.
 
-So, if `user` happens to be undefined due to a mistake, we'll know about it and fix it. Otherwise, coding errors can be silenced where not appropriate, and become more difficult to debug.
+Portanto, se `user` estiver indefinido devido a um erro, saberemos e o corrigiremos. Caso contrário, os erros poderão ser silenciados aonde não forem apropriados, e assim se tornando mais difíceis de debugar.
```

Comment:
In this context I think it's ok to translate the undefined. What do you think?

Reply:
Yes, I agree. In this context makes more sense to translate 'undefined' since it's not a code example